R/derive_param_confirmed_bor.R

```diff
@@ -0,0 +1,528 @@
+#'
+#' @author Stefan Bundfuss
+#'
+#' @export
+#'
+#' @examples
+#'
+#' library(dplyr)
+#'
+#' # Create ADSL dataset
+#' adsl <- tibble::tribble(
+#'   ~USUBJID, ~TRTSDTC,
+#'   "1",      "2020-01-01",
+#'   "2",      "2019-12-12",
+#'   "3",      "2019-11-11",
+#'   "4",      "2019-12-30",
+#'   "5",      "2020-01-01",
+#'   "6",      "2020-02-02",
+#'   "7",      "2020-02-02",
+#'   "8",      "2020-04-01"
+#' ) %>%
+#'   mutate(
+#'     TRTSDT = lubridate::ymd(TRTSDTC),
+#'     STUDYID = "XX1234"
+#'   )
+#'
+#' # Create ADRS dataset
+#' ovr_obs <- tibble::tribble(
+#'   ~USUBJID, ~ADTC,        ~AVALC,
+#'   "1",      "2020-01-01", "PR",
+#'   "1",      "2020-02-01", "CR",
+#'   "1",      "2020-02-16", "NE",
+#'   "1",      "2020-03-01", "CR",
+#'   "1",      "2020-04-01", "SD",
+#'   "2",      "2020-01-01", "SD",
+#'   "2",      "2020-02-01", "PR",
+#'   "2",      "2020-03-01", "SD",
+#'   "2",      "2020-03-13", "CR",
+#'   "3",      "2019-11-12", "CR",
+#'   "3",      "2019-12-02", "CR",
+#'   "3",      "2020-01-01", "SD",
+#'   "4",      "2020-01-01", "PR",
+#'   "4",      "2020-03-01", "SD",
+#'   "4",      "2020-04-01", "SD",
+#'   "4",      "2020-05-01", "PR",
+#'   "4",      "2020-05-15", "NON-CR/NON-PD",
+#'   "5",      "2020-01-01", "PR",
+#'   "5",      "2020-01-10", "SD",
+#'   "5",      "2020-01-20", "PR",
+#'   "5",      "2020-05-15", "NON-CR/NON-PD",
+#'   "6",      "2020-02-06", "PR",
+#'   "6",      "2020-02-16", "CR",
+#'   "6",      "2020-03-30", "PR",
+#'   "6",      "2020-04-12", "PD",
+#'   "6",      "2020-05-01", "CR",
+#'   "6",      "2020-06-01", "CR",
+#'   "7",      "2020-02-06", "PR",
+#'   "7",      "2020-02-16", "CR",
+#'   "7",      "2020-04-01", "NE"
+#' ) %>%
+#'   mutate(PARAMCD = "OVR")
+#'
+#' pd_obs <-
+#'   bind_rows(tibble::tribble(
+#'     ~USUBJID, ~ADTC,        ~AVALC,
+#'     "6",      "2020-04-12", "Y"
+#'   ) %>%
+#'     mutate(PARAMCD = "PD"))
+#' adrs <- bind_rows(ovr_obs, pd_obs) %>%
+#'   mutate(
+#'     ADT = lubridate::ymd(ADTC),
+#'     STUDYID = "XX1234"
+#'   ) %>%
+#'   select(-ADTC) %>%
+#'   admiral::derive_vars_merged(
+#'     dataset_add = adsl,
+#'     by_vars = dplyr::vars(STUDYID, USUBJID),
+#'     new_vars = dplyr::vars(TRTSDT)
+#'   )
+#'
+#' pd_date <- admiral::date_source(
+#'   dataset_name = "adrs",
+#'   date = ADT,
+#'   filter = PARAMCD == "PD"
+#' )
+#'
+#' # Derive confirmed best overall response parameter
+#' derive_param_confirmed_bor(
+#'   adrs,
+#'   dataset_adsl = adsl,
+#'   filter_source = PARAMCD == "OVR",
+#'   source_pd = pd_date,
+#'   source_datasets = list(adrs = adrs),
+#'   reference_date = TRTSDT,
+#'   ref_start_window = 28,
+#'   ref_confirm = 28,
+#'   set_values_to = vars(
+#'     PARAMCD = "CBOR",
+#'     PARAM = "Best Confirmed Overall Response by Investigator"
+#'   )
+#' ) %>%
+#'   filter(PARAMCD == "CBOR")
+#'
+#' # Derive confirmed best overall response parameter (accepting SD for PR and
+#' # considering missings as NE)
```

Comment:
maybe also include in this example allowing for 2 NEs in `max_nr_ne` and show that impact in an example patient

Reply:
Example extended.